simplesamlphp-module-privacyidea/templates/LoginForm.php

```diff
@@ -155,48 +155,45 @@ class="float-l erroricon" alt="gtk-dialog-error"/>
                                 <input id="password" name="password" tabindex="1" type="password" value="" class="text"
                                        placeholder="<?php echo htmlspecialchars($passHint, ENT_QUOTES) ?>"/>
 
-                                <strong id="message"><?php echo $this->data['message'] ?: "" ?></strong>
-
-                                <br><br>
-                                <label for="otp" class="sr-only">
-                                    <?php echo $this->t('{privacyidea:privacyidea:otp}'); ?>
-                                </label>
-
-                                <input id="otp" name="otp" tabindex="1" type="password" value="" class="text"
-                                       placeholder="<?php echo htmlspecialchars($otpHint, ENT_QUOTES) ?>"/>
+                                <strong id="message"><?php echo @$this->data['message'] ?: "" ?></strong>
+                                <br>
+                                <input id="otp" name="otp" type="password"
+                                       placeholder="<?php echo htmlspecialchars($otpHint, ENT_QUOTES) ?>">
                                 <br><br>
                                 <input id="submitButton" tabindex="1" class="rc-button rc-button-submit" type="submit"
                                        name="Submit"
                                        value="<?php echo htmlspecialchars($this->t('{login:login_button}'), ENT_QUOTES) ?>"/>
                                 <br><br>
 
-                                <!-- Hidden input which store the info about changes for future use in backend-->
+                                <!-- Undefined index is suppressed and the default is used for these values -->
                                 <input id="mode" type="hidden" name="mode"
-                                       value="<?php echo $this->data['mode'] ?: "otp" ?>"/>
+                                       value="<?php echo @$this->data['mode'] ?: "otp" ?>"/>
 
                                 <input id="pushAvailable" type="hidden" name="pushAvailable"
                                        value="<?php echo @$this->data['pushAvailable'] ?: false ?>"/>
```

Comment:
missing HTML escaping + better give a string as default, not boolean

Reply:
Because then the implicit PHP string conversion is not necessary - which only results on occasion the same as what you want here. You don't want to represent a PHP boolean in a HTML form but a specific value which is either `"false"`, "true"` or whatever.